src/server/server_family.cc

```diff
@@ -1749,29 +1753,20 @@ void ServerFamily::Hello(CmdArgList args, ConnectionContext* cntx) {
 
 void ServerFamily::ReplicaOfInternal(string_view host, string_view port_sv, ConnectionContext* cntx,
                                      ActionOnConnectionFail on_err) {
-  auto& pool = service_.proactor_pool();
   LOG(INFO) << "Replicating " << host << ":" << port_sv;
 
-  // We lock to protect global state changes that we perform during the replication setup:
-  // The replica_ pointer, GlobalState, and the DB itself (we do a flushall txn before syncing).
-  // The lock is only released during replica_->Start because we want to allow cancellation during
-  // the connection. If another replication command is received during Start() of an old
-  // replication, it will acquire the lock, call Stop() on the old replica_ and wait for Stop() to
-  // complete. So Replica::Stop() must
-  // 1. Be very responsive, as it is called while holding the lock.
-  // 2. Leave the DB in a consistent state after it is done.
-  // We have a relatively involved state machine inside Replica itself which handels cancellation
-  // with those requirements.
-  VLOG(2) << "Acquire replica lock";
+  // Wait until all transactions finished and prevent any other from executing
+  cntx->transaction->Schedule();
+  cntx->transaction->Execute([](auto*, auto*) { return OpStatus::OK; }, false);
```

Comment:
I'll transfer the comment from above. Technically it was wrong, because for example `We lock to protect global state changes that we perform during the replication setup: the replica_ pointer, GlobalState, and the DB itself`. We did _not_ lock the DB. We do now... But is there actually any point in it? We set the global state, so we have a clear boundary... 🤔